File: Excel Sheet Column Number.java
class Solution {
    public int titleToNumber(String s) {
        StringBuilder dummy = new StringBuilder(s) ;
        dummy.reverse() ;

        int result = 0 ;

        int pow = 1 ;

        for ( int i = 0 ; i < dummy.length() ; i++ ) {
            char c = dummy.charAt(i) ;
            result += pow * ((int)(c - 'A' + 1)) ;
            pow = pow * 26 ;
        }
        return result ;
    }
}

File: Excel Sheet Column Number.cpp
class Solution {
public:
    int titleToNumber(string s) {
        reverse(s.begin(), s.end()) ;

        int result = 0 ;

        long long int pow = 1 ;

        for ( int i = 0 ; i < s.length() ; i++ ) {
            char c = s[i] ;
            int k = (int)(c - 'A' + 1) ;
            result += (k * pow) ;
            pow = pow * 26 ;
        }

        return result ;
    }
};